### DISCUSSION TITLE
how to disable piping of stdout to less with i.e. git branch ?

### DISCUSSION THREAD

**Discussion by dulhaver** [ohmybash/oh-my-bash#563] [Q&A]
a non-oh-my-bash system `git branch` does roughly this:

```
user@vm-51150-0128:~/repos/ansible> git branch
  exe/prepare_postgres_workshop_230120
  feature/step_preps_generalized_230106
  mariadb_install_VM-413301-0001
* master
  master_copy_221213
  pgBackRest_step_install
gwagner@vm-51150-0128:~/repos/ansible>
```
so, it returns to stdout and provides a fresh prompt and you can proceed working.

with `oh-my-bash` this changes to some sort on `less-ish` output behaviour , meaning the stdout is piped to less.
```
user@vm-414001-0318 02:50:56 ~/repos/build_etcd |main U:2 ?:7 ✗| →  gb
```
```





  build_pg-15-postgis-3.2
  build_with_podman
* master
(END)
```
you need to leave that with the `q` to get back to a fresh prompt to continue working.

I find this not practical and am wondering how to switch this off in an `oh-my-bash` powered terminal?

**Comment by tyleralbee** [comment]
If I understand correctly, this is because of two things:

1. `git branch` [is piped to less](https://stackoverflow.com/questions/48341920/git-branch-command-behaves-like-less)
2. `less` is aliased to `alias less='less -FSRXc'` (see [aliases/general.aliases.sh](https://github.com/ohmybash/oh-my-bash/blob/0b218263fd456a6f3e7a84777385653f7070148a/aliases/general.aliases.sh#L66C1-L66C78))

I agree that for most people's use-case it's not practical for this to be the default behavior. You'd have to press `q` even if there was no output. I was thinking this was done so that you could have "hot reload" for commands like `git diff` but it doesn't seem like that works.

Is this meant to happen even when there's no output (e.g. from `git diff`)?

**Comment by dulhaver** [comment]
> Is this meant to happen even when there's no output (e.g. from git diff)?

it behaves like that, yes.

**Comment by Kreyren** [comment]
> how to disable piping of stdout to less with i.e. git branch ?

```console
$ git branch 1>/dev/null # Pipes stdout to /dev/null
```

https://www.cyberciti.biz/faq/how-to-redirect-output-and-errors-to-devnull/


Additionally you should just be able to pipe it even when it's doing `less`.
